salt/metalk8s/archives/unmounted.sls

```diff
@@ -0,0 +1,10 @@
+{% set removable_archives = salt.metalk8s.get_removable_archives() %}
+{% for env, archive in removable_archives.items() %}
+{% if archive.iso %}
+Archive {{ archive.iso }} absent at {{ archive.path }}:
+  mount.unmounted:
+    - name: {{ archive.path }}
+    - device: {{ archive.iso }}
+    - persist: true
+{% endif %}
```

Comment:
IMO we also want to remove the directory when we umount, only if the directory is empty

Something like that (not tested)
```yaml
Cleanup {{ archive.path }} directory:
  file.absent:
    - name: {{ archive.path }}
    - unless: file.path_exists_glob("{{ archive.path }}/*")
```